flink-runtime/src/test/java/org/apache/flink/runtime/concurrent/ManuallyTriggeredComponentMainThreadExecutor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.concurrent;
+
+/** Testing ComponentMainThreadExecutor which can be manually triggered. */
+public class ManuallyTriggeredComponentMainThreadExecutor
+        extends ManuallyTriggeredScheduledExecutorService implements ComponentMainThreadExecutor {
+
+    private Thread executorThread = null;
```

Comment:
Can we set this thread when creating the executor instead of setting it dynamically? We could simply call `assertRunningInMainThread` in the `trigger` method before calling `super.trigger()`.

Reply:
I meant to set this field when `ManuallyTriggeredComponentMainThreadExecutor` is being created. That way no random thread which calls `assertRunningInMainThread` first will be considered as the main thread.

Reply:
Damn. I was still thinking too complicated. For some reason I didn't realize that we are not really using a thread pool with different thread, but executing everything it the one main thread executing the tests.